Source/Core/DiscIO/VolumeVerifier.cpp

```diff
@@ -0,0 +1,648 @@
+    AddProblem(Severity::High, GetStringT("The data partition is missing."));
+
+  const bool has_channel_partition =
+      std::find(types.cbegin(), types.cend(), PARTITION_CHANNEL) != types.cend();
+  if (ShouldHaveChannelPartition() && has_channel_partition)
+    AddProblem(Severity::Medium, GetStringT("The channel partition is missing."));
+
+  if (ShouldHaveMasterpiecePartitions() &&
+      types.cend() ==
+          std::find_if(types.cbegin(), types.cend(), [](u32 type) { return type >= 0xFF; }))
+  {
+    // i18n: This string is referring to a game mode in Super Smash Bros. Brawl called Masterpieces
+    // where you play demos of NES/SNES/N64 games. Official translations:
+    // 名作トライアル (Japanese), Masterpieces (English), Meisterstücke (German), Chefs-d'œuvre
+    // (French), Clásicos (Spanish), Capolavori (Italian), 클래식 게임 체험판 (Korean).
+    // If your language is not one of the languages above, consider leaving the string untranslated
+    // so that people will recognize it as the name of the game mode.
+    AddProblem(Severity::Medium, GetStringT("The Masterpiece partitions are missing."));
+  }
+
+  for (const Partition& partition : partitions)
+  {
+    if (m_volume.GetPartitionType(partition) == std::optional<u32>(PARTITION_UPDATE) &&
+        partition.offset != 0x50000)
+    {
+      AddProblem(Severity::Low, GetStringT("The update partition is not at its normal position."));
+    }
+
+    const u64 normal_data_offset = IsDebugSigned() ? 0x838000 : 0xF800000;
+    if (m_volume.GetPartitionType(partition) == std::optional<u32>(PARTITION_DATA) &&
+        partition.offset != normal_data_offset && !has_channel_partition)
+    {
+      AddProblem(
+          Severity::Low,
+          GetStringT("The data partition is not at its normal position. This will affect the "
+                     "emulated loading times. When using netplay or sending input recordings to "
+                     "other people, you will experience desyncs if anyone is using a good dump."));
+    }
+  }
+
+  std::vector<Partition> partitions_result;
+  for (const Partition& partition : partitions)
+  {
+    if (CheckPartition(partition))
+      partitions_result.emplace_back(partition);
+  }
+
+  return partitions;
+}
+
+bool VolumeVerifier::CheckPartition(const Partition& partition)
+{
+  std::optional<u32> type = m_volume.GetPartitionType(partition);
+  if (!type)
+  {
+    // Not sure if this can happen in practice
+    AddProblem(Severity::Medium, GetStringT("The type of a partition could not be read."));
+    return false;
+  }
+
+  Severity severity = Severity::Medium;
+  if (*type == PARTITION_DATA)
+    severity = Severity::High;
+  else if (*type == PARTITION_UPDATE)
+    severity = Severity::Low;
+
+  std::string name = NameForPartitionType(*type, false);
+  if (ShouldHaveMasterpiecePartitions() && *type > 0xFF)
+  {
+    // i18n: This string is referring to a game mode in Super Smash Bros. Brawl called Masterpieces
```

Comment:
Hm, this string is singular, but the description is plural. Should we explicitly point out that the string uses the singular form "Masterpiece" instead of the plural "Masterpieces"?

Reply:
This comment already points that out.

Reply:
Duh, my bad. Only skimmed it, and start/end being similar fooled me.